clients/unshaded/src/main/java/tachyon/client/file/FileSystem.java

```diff
@@ -0,0 +1,375 @@
+   * @throws FileDoesNotExistException if the given path does not exist
+   * @throws TachyonException if an unexpected Tachyon exception is thrown
+   */
+  List<URIStatus> listStatus(TachyonURI path)
+      throws FileDoesNotExistException, IOException, TachyonException;
+
+  /**
+   * If the path is a directory, returns the {@link URIStatus} of all the direct entries in it.
+   * Otherwise returns the {@link URIStatus} for the file.
+   *
+   * @param path the path to list information about
+   * @param options options to associate with this operation
+   * @return a list of {@link URIStatus}s containing information about the files and directories
+   *         which are children of the given path
+   * @throws IOException if a non-Tachyon exception occurs
+   * @throws FileDoesNotExistException if the given path does not exist
+   * @throws TachyonException if an unexpected Tachyon exception is thrown
+   */
+  List<URIStatus> listStatus(TachyonURI path, ListStatusOptions options)
+      throws FileDoesNotExistException, IOException, TachyonException;
+
+  /**
+   * Convenience method for {@link #loadMetadata(TachyonURI, LoadMetadataOptions)} with default
+   * options.
+   *
+   * @param path the path for which to load metadata from UFS
+   * @return the {@link TachyonURI} referencing the new path available in Tachyon
+   * @throws IOException if a non-Tachyon exception occurs
+   * @throws FileDoesNotExistException if the given path does not exist
+   * @throws TachyonException if an unexpected Tachyon exception is thrown
+   */
+  TachyonURI loadMetadata(TachyonURI path)
+      throws FileDoesNotExistException, IOException, TachyonException;
+
+  /**
+   * Loads metadata about a path in the UFS to Tachyon. No data will be transferred.
+   *
+   * @param path the path for which to load metadata from UFS
+   * @param options options to associate with this operation
+   * @return the {@link TachyonURI} referencing the new path available in Tachyon
+   * @throws IOException if a non-Tachyon exception occurs
+   * @throws FileDoesNotExistException if the given path does not exist
+   * @throws TachyonException if an unexpected Tachyon exception is thrown
+   */
+  TachyonURI loadMetadata(TachyonURI path, LoadMetadataOptions options)
+      throws FileDoesNotExistException, IOException, TachyonException;
+
+  /**
+   * Convenience method for {@link #mount(TachyonURI, TachyonURI, MountOptions)} with default
+   * options.
+   *
+   * @param src a Tachyon path to mount the data to
+   * @param dst a UFS path to mount the data from
+   * @throws IOException if a non-Tachyon exception occurs
+   * @throws TachyonException if a Tachyon exception occurs
+   */
+  void mount(TachyonURI src, TachyonURI dst) throws IOException, TachyonException;
+
+  /**
+   * Mounts a UFS subtree to the given Tachyon path. The Tachyon path is expected not to exist as
+   * the method creates it. This method does not transfer any data or metadata from the UFS. It
```

Comment:
Due to lazy loading you still would not get any additional data. you would need to call getStatus(src/file) to load that data, or simply loadmetadata(src)